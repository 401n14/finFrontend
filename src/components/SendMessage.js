import React from 'react';
/**
 * Component for sending messages
 * Input field for users to enter a message
 * Button to send message
 * @component SendMessage
 * @param {object} props
 * @return {ReactElement} HTML markup
 */
function SendMessage(props) {
  return (
    <div>
      <div className='chat-section'>
        <input
          type='text'
          className='chat-input, chat-typing input-text'
          value={props.value}
          onKeyUp={props.onKeyUp}
          onChange={props.onChange}
        />
        <button className='btn btn-secondary chat-btn' onClick={props.onClick}>
          Send Message
        </button>
        <svg
          className="chat-svg"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 20 20"
          fill='red'
        >
          <path d="M18 12v1H8v5l-6-6 6-6v5h8V2h2z" />
        </svg>
      </div>
    </div>
  );
}

export default SendMessage;
