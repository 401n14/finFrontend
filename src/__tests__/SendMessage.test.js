import React from 'react';
import SendMessage from '../components/SendMessage';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


configure({adapter:new Adapter()});

describe('Snapshot Test', () => {
    it('renders correctly, Snapshot', () => {
        const page = renderer.create(<SendMessage />).toJSON();
        expect(page).toMatchSnapshot();
    });
});