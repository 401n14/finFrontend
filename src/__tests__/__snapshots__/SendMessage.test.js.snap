// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly, Snapshot 1`] = `
<div>
  <div
    className="chat-section"
  >
    <input
      className="chat-input, chat-typing input-text"
      type="text"
    />
    <button
      className="btn btn-secondary chat-btn"
    >
      Send Message
    </button>
    <svg
      className="chat-svg"
      fill="red"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 12v1H8v5l-6-6 6-6v5h8V2h2z"
      />
    </svg>
  </div>
</div>
`;
